File: dayOne/java/levelOne/SplitArrayTest.java
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertArrayEquals;

public class SplitArrayTest {

    @Test
    public void testThatIPassAnArrayAndItReturnsAnArrayOfArraysWithEvenAndOddNumbersSeparated() {

        int[] array = {45, 60, 3, 10, 9, 22};

        int[][] actual = SplitArray.splitArray(array);
        int[][] expected = {{45, 3, 9}, {60, 10, 22}};

        assertArrayEquals(expected, actual);
    }

    @Test
    public void testThatIPassAnArrayWithOnlyOddNumbersAndItReturnsAnArrayOfArraysWithEmptyEvenAndOddNumbersSeparated() {

        int[] array = {45, 3, 5, 11, 31, 9};

        int[][] actual = SplitArray.splitArray(array);
        int[][] expected = {{45, 3, 5, 11, 31, 9}, {}};

        assertArrayEquals(expected, actual);
    }

    @Test
    public void testThatIPassAnArrayWithOnlyEvenNumbersAndItReturnsAnArrayOfArraysWithEvenAndEmptyOddNumbersSeparated() {

        int[] array = {40, 24, 32, 64, 58};

        int[][] actual = SplitArray.splitArray(array);
        int[][] expected = {{}, {40, 24, 32, 64, 58}};

        assertArrayEquals(expected, actual);
    }
}

File: dayOne/javaScript/levelOne/splitArray.test.js
const { splitArray } = require("./splitArray");

test("I pass an array and it split it into two arrays of even and odds", () => {

    let numbers = [45, 60, 3, 10, 9, 22];

    let expected = [[45, 3, 9], [60, 10, 22]];

    let actual = splitArray(numbers);

    expect(actual).toEqual(expected);
});

test("I pass an array of odd numbers and it split it into two arrays of empty even and odds",
    () => {

    let numbers = [45, 61, 3, 11, 9, 23];

    let expected = [[45, 61, 3, 11, 9, 23], []];

    let actual = splitArray(numbers);

    expect(actual).toEqual(expected);
});

test("I pass an array of even numbers and it split it into two arrays of even and empty odds",
    () => {

    let numbers = [46, 60, 4, 10, 16, 22];

    let expected = [[], [46, 60, 4, 10, 16, 22]];

    let actual = splitArray(numbers);

    expect(actual).toEqual(expected);
});
